dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java

```diff
@@ -62,10 +65,18 @@ protected final void service(HttpServletRequest req, HttpServletResponse respons
                 return;
             }
 
-            request.setRequestUri(uri);
             final PageMode mode = PageMode.getWithNavigateMode(request);
             try {
-                VelocityModeHandler.modeHandler(mode, request, response).serve();
+                final String pageHtml = APILocator.getHTMLPageAssetRenderedAPI().getPageHtml(
+                        PageContextBuilder.builder()
+                                .setPageUri(uri)
+                                .setPageMode(mode)
+                                .build(),
+                        request,
+                        response
+                );
+
+                response.getOutputStream().write(pageHtml.getBytes());
```

Comment:
This is a very scary fix to me.  Are we sure that responses from the LIVE|PREVIEW|EDIT ModeHandlers are a one to one match with the `HTMLPageAssetRenderedAPI`?  Does the `APILocator.getHTMLPageAssetRenderedAPI()` respect the same caching mechanisms that are being used in the VelocityModeHandlers?  Have we tried any benchmarking using this pattern?  Are there URI cases that are not going to work?

This might be a right change but it really needs to be proved out.